gammapy/maps/wcs.py

```diff
@@ -459,6 +459,45 @@ def footprint(self):
         coords = self.wcs.calc_footprint()
         return SkyCoord(coords, frame=self.frame, unit="deg")
 
+    @classmethod
+    def from_aligned(cls, geom, skydir, width):
+        """Create an aligned geometry from an existing one
+
+        Parameters
+        ----------
+        geom : `~WcsGeom`
+            A reference WCS geometry object.
+        skydir : tuple or `~astropy.coordinates.SkyCoord`
+            Sky position of map center.  Can be either a SkyCoord
+            object or a tuple of longitude and latitude in deg in the
+            coordinate system of the map.
+        width : float or tuple or list or string
+            Width of the map in degrees.  A tuple will be interpreted
+            as parameters for longitude and latitude axes.  For maps
+            with non-spatial dimensions, list input can be used to
+            define a different map width in each image plane.
+
+        Returns
+        -------
+        geom : `~WcsGeom`
+            An aligned WCS geometry object with specified size and center.
+
+        """
+        width = _check_width(width) * u.deg
+        xref, yref = geom.to_image().coord_to_idx(skydir)
+        npix = np.rint(width / geom.pixel_scales).astype(int)
+        xref = -int(xref) + npix[0] // 2 + geom.wcs.wcs.crpix[0]
+        yref = -int(yref) + npix[1] // 2 + geom.wcs.wcs.crpix[1]
+        return cls.create(
+            skydir=tuple(geom.wcs.wcs.crval),
+            npix=tuple(npix),
+            refpix=(xref - 1, yref),
```

Comment:
Honestly I have no idea why the `-1` here is required. I added it so the resulting geometry is equivalent to one returned by `.cutout()`. See the [test vs cutout](https://github.com/gammapy/gammapy/pull/3399/files#diff-33810c8943a3a35af445d931195c7b17ae552ac859c334dedd1f502b7c14ccedR155). I think this is non-ideal, however the behaviour is established with a test, so possibly OK to keep for now...